### ISSUE TITLE
Failed dockerfile build @ samtools-cloud and possible solution

### ISSUE THREAD

**Issue by PLTJINXX** [broadinstitute/gatk-sv#571]
When building dockerfiles/samtools-cloud, I encountered this issue:

```
#8 0.581 Hit:1 http://security.ubuntu.com/ubuntu jammy-security InRelease
#8 0.590 Hit:2 http://archive.ubuntu.com/ubuntu jammy InRelease
#8 0.636 Hit:3 http://archive.ubuntu.com/ubuntu jammy-updates InRelease
#8 0.685 Hit:4 http://archive.ubuntu.com/ubuntu jammy-backports InRelease
#8 0.772 Get:5 http://packages.cloud.google.com/apt cloud-sdk InRelease [6361 B]
#8 1.014 Err:5 http://packages.cloud.google.com/apt cloud-sdk InRelease
#8 1.014   The following signatures couldn't be verified because the public key is not available: NO_PUBKEY B53DC80D13EDEF05
#8 1.020 Reading package lists...
#8 1.750 W: GPG error: http://packages.cloud.google.com/apt cloud-sdk InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY B53DC80D13EDEF05
#8 1.750 E: The repository 'http://packages.cloud.google.com/apt cloud-sdk InRelease' is not signed.
#8 ERROR: executor failed running [/bin/sh -c echo "deb [signed-by=/usr/share/keyrings/cloud.google.gpg] http://packages.cloud.google.com/apt cloud-sdk main"     | tee -a /etc/apt/sources.list.d/google-cloud-sdk.list &&     curl https://packages.cloud.google.com/apt/doc/apt-key.gpg |     tee /usr/share/keyrings/cloud.google.gpg &&     apt-get update -y &&     apt-get install google-cloud-sdk=$CLOUD_SDK_VERSION -y &&     gcloud config set core/disable_usage_reporting true &&     gcloud config set component_manager/disable_update_check true &&     gcloud config set metrics/environment github_docker_image]: exit code: 100
```

Replacing the run command for building the gcloud CLI (as shown below) fixes the issue with no hindrance to the pipeline thus far: 
File: dockerfiles/samtools-cloud/Dockerfile

```
RUN echo "deb [signed-by=/usr/share/keyrings/cloud.google.gpg] http://packages.cloud.google.com/apt cloud-sdk main" \
    | tee -a /etc/apt/sources.list.d/google-cloud-sdk.list && \
    curl https://packages.cloud.google.com/apt/doc/apt-key.gpg \
    | apt-key --keyring /usr/share/keyrings/cloud.google.gpg  add - && \
    apt-get update -y && apt-get install google-cloud-sdk=$CLOUD_SDK_VERSION -y && \
    gcloud config set core/disable_usage_reporting true && \
    gcloud config set component_manager/disable_update_check true && \
    gcloud config set metrics/environment github_docker_image
```

Is this a valid solution? The dockerfile is being built on a MacOS.

**Comment by mwalker174** [comment]
That solution might work, but I wonder if just rebuilding `samtools-cloud-env` would update the keys as well. @VJalili

**Comment by VJalili** [comment]
@PLTJINXX thanks for reporting this issue. Are you building the image on a Mac with an M1 chip?

**Comment by PLTJINXX** [comment]
No, my Mac does not have the M1 chip. It's an older model.

**Comment by VJalili** [comment]
@PLTJINXX, I was able to reproduce the issue on an Ubuntu VM. Using the `apt-key --keyring` as you suggested aligns with the updated GCP documentation, and it results in successfully building the docker image. #574 fixes this bug. Thanks for reporting it.

**Comment by VJalili** [comment]
@PLTJINXX Hopefully the issue is resolved in #574, so I am closing this issue, but please feel free to reopen it if you experience a similar issue.
